### PR TITLE
Add outcome coverage for a class to a course

### PR DESCRIPTION
**Pull request by margaretoti**
[Trello Card](https://trello.com/c/UxUB8m5r)

This branch adds outcome coverage to a subject/class. Outcome coverage refers to the one or more outcomes covered by any given subject/class in a course. Connecting outcomes to subjects/classes and then subjects/classes to courses means that the UI can show a user which outcomes have been matched to a class in the course and which outcomes still need to be demonstrated by a subject/class in the course.

### CODE REVIEW

File: app/models/outcome_coverage.rb [hunk 1]
```diff
@@ -0,0 +1,8 @@
+class OutcomeCoverage < ActiveRecord::Base
+  belongs_to :course
+  belongs_to :outcome
+  belongs_to :subject
+
+  delegate :title, to: :subject, prefix: true
+  delegate :name, :nickname, to: :outcome, prefix: true
```
Comment (geoffharcourt) [inline app/models/outcome_coverage.rb]: worth :abc:-ing these by the thing they delegate to?

File: config/routes.rb [hunk 2]
```diff
@@ -9,6 +9,7 @@
     root "dashboard#show"
 
     resources :courses, only: [:index] do
+      resources :outcome_coverages, only: [:new, :create, :index]
```
Comment (geoffharcourt) [inline config/routes.rb]: :abc: these? (I have some trauma from my last app having a very long routes list).

File: spec/features/user_adds_outcome_coverage_to_a_course_spec.rb [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    subject = create(:subject)
+    course = create(:course)
+    outcome = create(:outcome, course: course)
+    user = user_with_assessments_access_to(course.department)
+
+    visit manage_assessments_course_outcome_coverages_path(course, as: user)
+    click_on I18n.t('manage_assessments.outcome_coverages.index.add_a_class')
+    select subject.title, from: "outcome_coverage[subject_id]"
+    select outcome.nickname, from: "outcome_coverage[outcome_id]"
+    click_button I18n.t('helpers.submit.outcome_coverage.create')
```
Comment (geoffharcourt) [inline spec/features/user_adds_outcome_coverage_to_a_course_spec.rb]: I18n is included in tests, so I think you can just do `t(...)` here.

https://github.com/MIT-IR/abet/blob/master/spec/support/I18n.rb

File: app/views/manage_assessments/outcome_coverages/new.html.erb [hunk 4]
```diff
@@ -0,0 +1,5 @@
+<%= simple_form_for @outcome_coverage, url: manage_assessments_course_outcome_coverages_path(course_id: @outcome_coverage.course.id) do |form| %>
```
Comment (derekprior) [inline app/views/manage_assessments/outcome_coverages/new.html.erb]: I think we can just do: `manage_assessments_course_outcome_coverages_path(@outcome_coverage.course)`

Comment (derekprior) [inline app/views/manage_assessments/outcome_coverages/new.html.erb]: woah? really. Sounds interesting. Disregard for now then.

Comment (margaretoti) [inline app/views/manage_assessments/outcome_coverages/new.html.erb]: Sure. Will do.

**Pull request merged**